Lib/test/test_math.py

```diff
@@ -1803,14 +1803,14 @@ def test_prod(self):
         self.assertRaises(TypeError, prod)
         self.assertRaises(TypeError, prod, 42)
         self.assertRaises(TypeError, prod, ['a', 'b', 'c'])
-        self.assertRaises(TypeError, prod, ['a', 'b', 'c'], '')
-        self.assertRaises(TypeError, prod, [b'a', b'c'], b'')
+        self.assertRaises(TypeError, prod, ['a', 'b', 'c'], start='')
+        self.assertRaises(TypeError, prod, [b'a', b'c'], start=b'')
         values = [bytearray(b'a'), bytearray(b'b')]
-        self.assertRaises(TypeError, prod, values, bytearray(b''))
+        self.assertRaises(TypeError, prod, values, start=bytearray(b''))
         self.assertRaises(TypeError, prod, [[1], [2], [3]])
         self.assertRaises(TypeError, prod, [{2:3}])
-        self.assertRaises(TypeError, prod, [{2:3}]*2, {2:3})
-        self.assertRaises(TypeError, prod, [[1], [2], [3]], [])
+        self.assertRaises(TypeError, prod, [{2:3}]*2, start={2:3})
+        self.assertRaises(TypeError, prod, [[1], [2], [3]], start=[])
         with self.assertRaises(TypeError):
             prod([10, 20], [30, 40])     # start is a keyword-only argument
```

Comment:
It looks not clear.
```suggestion
            prod([10, 20], 1)     # start is a keyword-only argument
```

Reply:
You can add also tests for weird cases like `prod([2, 3], start='ab')`, `prod([2, 3], start=[1, 2])` and `prod([], start={2: 3})`.

Reply:
> You can add also tests for weird cases like `prod([2, 3], start='ab')`, `prod([2, 3], start=[1, 2])` and `prod([], start={2: 3})`.

Done!